cirq/google/engine/engine.py

```diff
@@ -707,6 +726,103 @@ def get_metrics_by_name(self, name: str) -> List[Dict]:
         return result
 
 
+class EngineProgram:
+    """A program created via the Quantum Engine API.
+
+    This program wraps a Circuit or Schedule with additional metadata used to
+    schedule againt devices maintained by Google.
+
+    Attributes:
+      name: The full resource name of the engine program.
+      code: A serialized version of the Circuit or Schedule
+    """
+
+    def __init__(self, program: Dict[str, Any], engine: Engine) -> None:
+        """A job submitted to the engine.
+
+        Args:
+            program: A full Program Dict from the Engine.
```

Comment:
I think you only use the program resource id?  I'd recommend just using that unless there is another use of the program dict

Reply:
My thought was that this could scale to do more without changing the interface, but fine by me to keep it simple for now. \Updated.